src/components/BatchAllocations/hooks/useSourceData/useSourceData.js

```diff
@@ -0,0 +1,36 @@
+import { useQuery } from 'react-query';
+
+import {
+  useNamespace,
+  useOkapiKy,
+} from '@folio/stripes/core';
+
+import {
+  LEDGERS_API,
+  GROUPS_API,
+  BATCH_ALLOCATIONS_SOURCE,
+} from '../../../../common/const';
+
+export const useSourceData = (source, id) => {
+  const ky = useOkapiKy();
+  const [namespace] = useNamespace({ key: source });
+  const api = source === BATCH_ALLOCATIONS_SOURCE.ledger ? LEDGERS_API : GROUPS_API;
+
+  const {
+    data = {},
+    isFetching,
+    isLoading,
+  } = useQuery(
+    [namespace, id],
+    async ({ signal }) => ky.get(`${api}/${id}`, { signal }).json(),
+    {
+      enabled: Boolean(id),
+    },
+  );
```

Comment:
could you please pass an object of params instead
e.g. https://github.com/folio-org/ui-finance/blob/master/src/common/hooks/useOrder/useOrder.js#L14-L21